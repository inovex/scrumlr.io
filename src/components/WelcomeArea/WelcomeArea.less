@import '../../constants/style';

.welcome-area {
  position: relative;
  display: flex;
  flex-wrap: wrap;

  overflow: auto;

  width: 100vw;
  height: 100vh;
}

.welcome-area__headline {
  background: url('../../assets/logo-scrumlr-beta.svg') no-repeat;
  background-position-x: center;

  height: 74px;
  width: 164px;
  text-indent: -10000px;

  margin-bottom: @large-margin*2;
}

.welcome-area__promotion-wrapper {
  background: url('../../assets/visual-chair-scrumlr.svg') no-repeat @large-margin 72px, linear-gradient(to bottom, @color-mint, @color-darkest-mint);
  background-size: auto 100%;

  flex: 2 1 66%;

  display: flex;
  flex-direction: column;
  justify-content: center;
}

.welcome-area__promotion {
  font-size: initial;
  text-align: left;

  min-width: 320px;
  max-width: 640px;

  align-self: center;

  padding: @large-padding;
}

.welcome-area__promotion-text > p {
  color: @color-white;
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
  line-height: 24px;
}

.welcome-area__promotion-text > h2 {
  color: @color-white;
  font-weight: 900;
  font-size: 6vw;
  letter-spacing: 0;
  line-height: 8vw;
}

.welcome-area__action-area {
  display: flex;
  flex-direction: column;
  justify-content: center;

  flex: 1 1 33%;

  background-color: #fff;
  padding: @large-padding*2;
}

@media @tablet {
  .welcome-area__promotion-text > h2 {
    font-size: 36px;
    letter-spacing: 0;
    line-height: 48px;
  }

  .welcome-area {
    flex-wrap: nowrap;
  }

  .welcome-area__action-area {
    min-width: 300px;
    max-width: 420px;
  }

  .welcome-area__action-area {
    z-index:1;
    box-shadow: 0 0 @base-unit*16 0 fade(@color-dark, 65%);
  }
}


