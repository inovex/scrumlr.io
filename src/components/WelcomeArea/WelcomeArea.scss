@import '../../constants/style';

.welcome-area {
  position: relative;
  display: flex;
  flex-wrap: wrap;

  overflow: auto;

  width: 100vw;
  min-height: 100vh;
}

.welcome-area__headline {
  background: url('../../assets/logo-scrumlr-beta.svg') no-repeat;
  background-position-x: center;

  height: 74px;
  width: 164px;
  text-indent: -10000px;

  margin-bottom: $large-margin*2;
}

.welcome-area__promotion-wrapper {
  background: url('../../assets/visual-chair-scrumlr.svg') no-repeat $large-margin 72px, linear-gradient(to bottom, #0057ff, #5086ef);
  background-size: auto 100%;

  flex: 2 1 66%;

  display: flex;
  flex-direction: column;
  justify-content: center;
}

.welcome-area__promotion {
  font-size: initial;
  text-align: left;

  min-width: 320px;
  max-width: 640px;

  align-self: center;

  padding: $large-padding $large-padding 64px;
}

.welcome-area__promotion-text > p {
  color: $color-white;
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
  line-height: 24px;
}

.welcome-area__promotion-text > h2 {
  color: $color-white;
  font-weight: 900;
  font-size: 6vw;
  letter-spacing: 0;
  line-height: 8vw;
}

.welcome-area__action-area {
  display: flex;
  flex-direction: column;
  justify-content: center;

  flex: 1 1 33%;

  background-color: #fff;
  padding: $large-padding;
}

.welcome-area__action-area-main {
  display: flex;
  flex-direction: column;
  justify-content: center;

  flex: 1 1 100%;
}

.welcome-area__action-area-footer {
  text-align: center;

  font-size: 13.6px;
  color: $color-stone-gray;
}

.welcome-area__action-area-footer p {
  margin-bottom: .25em;
}

.welcome-area__action-area-footer a {
  font-weight: bold;
  color: currentColor;
}

.welcome-area__action-area-logo {
  vertical-align: middle;
}

.welcome-area__policies ul {
  list-style: none;
  padding: 0;
  margin: .75em 0 0;
}

.welcome-area__policies ul > li {
  display: inline;
  white-space: nowrap;
}

.welcome-area__policies ul > li:before {
  content: " | ";
}

.welcome-area__policies ul > li:first-child:before {
  content: '';
}

@media #{$tablet} {
  .welcome-area__promotion-text > h2 {
    font-size: 36px;
    letter-spacing: 0;
    line-height: 48px;
  }

  .welcome-area {
    flex-wrap: nowrap;
  }

  .welcome-area__action-area {
    min-width: 300px;
    max-width: 420px;

    padding: $large-padding*2;
  }

  .welcome-area__action-area {
    z-index:1;
    box-shadow: 0 0 $base-unit*16 0 transparentize($color-dark, .45);
  }
}

.welcome-area__migration {
  background: #ffebee;
  border-radius: 6px;
  border: 1px solid red;
  padding: 16px 16px 42px;

  position: relative;
}

.welcome-area__migration-disclaimer {
  font-weight: bold;
}

.welcome-area__gimme-beta {
  position: absolute;
  bottom: -32px;
  min-width: 240px;
  left: 50%;
  transform: translateX(-50%);

  display: flex;
  align-items: center;
  justify-content: center;



  padding: 16px 32px;
  background: #d03d3d;
  color: white;
  border: none;
  font-weight: 900;
  border-radius: 6px;

  text-align: center;
  text-decoration: none;

  transition: background-color 0.1s ease-in-out;
}

.welcome-area__gimme-beta:hover,
.welcome-area__gimme-beta:focus, {
  background: #ff0000;
}

.welcome-area__gimme-beta-icon {
  margin-left: 16px;
}

.welcome-area__gimme-beta-icon * {
  fill: #ffffff;
}

