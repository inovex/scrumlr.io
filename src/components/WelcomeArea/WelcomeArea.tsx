import * as React from 'react';
import './WelcomeArea.css';

export const WelcomeArea: React.SFC<{}> = ({ children }) =>
  <div className="welcome-area">
    <div className="welcome-area__promotion-wrapper">
      <div className="welcome-area__promotion">
        <h1 className="welcome-area__headline">Scrumlr</h1>

        <div className="welcome-area__promotion-text">
          <h2>Progress is impossible without change</h2>

          <p>
            Take a seat and let us guide you and your team through your first
            online retrospective. Start your first session within seconds -
            completely for free, no registration or payment details required.
            Put your thoughts and your feedback on cards and share them
            instantly on a collaborative virtual board. Vote on the topics that
            really matter. Learn and grow as a team.
          </p>
        </div>
      </div>
    </div>

    <div className="welcome-area__action-area">
      {children}
    </div>
  </div>;

export default WelcomeArea;
