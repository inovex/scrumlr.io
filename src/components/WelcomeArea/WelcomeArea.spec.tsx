import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import WelcomeArea from './WelcomeArea';

describe('<WelcomeArea />', () => {
  it('should match snapshot', () => {
    const wrapper: ShallowWrapper<{}, {}> = shallow(
      <WelcomeArea>Some content</WelcomeArea>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
