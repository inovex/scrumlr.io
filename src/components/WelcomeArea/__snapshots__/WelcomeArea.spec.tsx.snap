// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeArea /> should match snapshot 1`] = `"<div class=\\"welcome-area\\"><div class=\\"welcome-area__promotion-wrapper\\"><div class=\\"welcome-area__promotion\\"><h1 class=\\"welcome-area__headline\\">Scrumlr</h1><div class=\\"welcome-area__promotion-text\\"><h2>Progress is impossible without change</h2><p>Take a seat and let us guide you and your team through your first online retrospective. Start your first session within seconds - completely for free, no registration or payment details required. Put your thoughts and your feedback on cards and share them instantly on a collaborative virtual board. Vote on the topics that really matter. Learn and grow as a team.</p></div></div></div><div class=\\"welcome-area__action-area\\">Some content</div></div>"`;
